src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -66,47 +65,57 @@ namespace ts.codefix {
                 }
 
                 return [{
-                    description: formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.Initialize_property_0_in_the_constructor), [memberName]),
+                    description: formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.Initialize_property_0_in_the_constructor), [tokenName]),
                     changes: [{
                         fileName: sourceFile.fileName,
                         textChanges: [{
                             span: { start: classConstructor.body.getEnd() - 1, length: 0 },
-                            newText: `this.${memberName} = undefined;${context.newLineCharacter}`
+                            newText: `this.${tokenName} = undefined;${context.newLineCharacter}`
                         }]
                     }]
                 }];
             }
         }
 
         function getActionsForAddMissingMemberInTypeScriptFile(): CodeAction[] | undefined {
-            let typeNode: TypeNode;
+            const openBrace = getOpenBraceOfClassLike(classDeclaration, sourceFile);
+            let actions: CodeAction[];
+
+            if (token.parent.parent.kind === SyntaxKind.CallExpression) {
+                const callExpression = <CallExpression>token.parent.parent;
+                const methodDeclaration = createMethodFromCallExpression(callExpression, tokenName);
+
+                const methodDeclarationChangeTracker = textChanges.ChangeTracker.fromCodeFixContext(context);
+                methodDeclarationChangeTracker.insertNodeAfter(sourceFile, openBrace, methodDeclaration, { suffix: context.newLineCharacter });

```

Comment:
also add handling for JS functions. basically the same but no type annotation or type parameters.